pkg/virt-dhcp/dnsmasq.go

```diff
@@ -0,0 +1,240 @@
+package virtdhcp
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"syscall"
+
+	"kubevirt.io/kubevirt/pkg/logging"
+)
+
+const (
+	DNSmasqExec   = "/usr/sbin/dnsmasq"
+	DHCPConfig    = "/tmp/dhcp.conf"
+	DHCPLeaseFile = "/tmp/dhcp.leases"
+	DHCPPidFile   = "/tmp/dhcp.pid"
+)
+
+type DNSmasqInstance struct {
+	hosts    map[string]string
+	ipRange  [][]byte
+	confpath string
+	args     []string
+	cmd      *exec.Cmd
+}
+
+func NewDNSmasq() *DNSmasqInstance {
+	dnsMasq := new(DNSmasqInstance)
+	dnsMasq.hosts = make(map[string]string)
+	dnsMasq.ipRange = make([][]byte, 2)
+	dnsMasq.args = []string{"-k", "-d", "--conf-file=" + DHCPConfig, "--dhcp-leasefile=" + DHCPLeaseFile, "--pid-file=" + DHCPPidFile}
+	return dnsMasq
+}
+
+// implement `Interface` in sort package.
+type sortByteArrays [][]byte
+
+func (b sortByteArrays) Len() int {
+	return len(b)
+}
+
+func (b sortByteArrays) Less(i, j int) bool {
+	// bytes package already implements Comparable for []byte.
+	switch bytes.Compare(b[i], b[j]) {
+	case -1:
+		return true
+	case 0, 1:
+		return false
+	default:
+		return false
+	}
+}
+
+func (b sortByteArrays) Swap(i, j int) {
+	b[j], b[i] = b[i], b[j]
+}
+
+func (dnsmasq *DNSmasqInstance) Start() error {
+	logging.DefaultLogger().Debug().Msg("starting Dnsmasq")
+	cmd := exec.Command(DNSmasqExec, dnsmasq.args...)
+
+	dnsmasq.cmd = cmd
+	err := cmd.Start()
+	if err != nil {
+		logging.DefaultLogger().Error().Reason(err).Msg("failed to start dnsmasq")
+		return err
+	}
+	go func(cmd *exec.Cmd) {
+		err := cmd.Wait()
+		if err != nil {
+			logging.DefaultLogger().Warning().Reason(err).Msgf("dnsmasq process exit")
+		}
+	}(cmd)
+	return nil
+}
+
+func checkProcessExist(pid int) (bool, *os.Process) {
+	proc, err := os.FindProcess(pid)
+	if err != nil {
+		logging.DefaultLogger().Warning().Msgf("failed to find process %d", pid)
+		return false, nil
+	}
+	procErr := proc.Signal(syscall.Signal(0))
+	return procErr == nil, proc
+}
+
+func (dnsmasq *DNSmasqInstance) stop() error {
+	logging.DefaultLogger().Info().Msg("Stopping Dnsmasq")
+	if dnsmasq.cmd != nil {
+		procExist, proc := checkProcessExist(dnsmasq.cmd.Process.Pid)
+		if procExist {
+			logging.DefaultLogger().Debug().Msgf("killing process %d", proc.Pid)
+			killErr := proc.Kill()
+			if killErr != nil {
+				logging.DefaultLogger().Warning().Msgf("failed to kill process %d", proc.Pid)
+			}
+
+		}
+	} else {
+		if _, err := os.Stat(DHCPPidFile); !os.IsNotExist(err) {
+			pid, err1 := getPidFromFile(DHCPPidFile)
+			if err1 == nil {
+				procExist, proc := checkProcessExist(pid)
+				if procExist {
+					logging.DefaultLogger().Debug().Msgf("killing process %d", proc.Pid)
+					proc.Kill()
+				}
+			}
+
+		}
+	}
+
+	return nil
+}
+
+func (dnsmasq *DNSmasqInstance) Restart() error {
+	logging.DefaultLogger().Info().Msg("restarting Dnsmasq")
+
+	err := dnsmasq.stop()
```

Comment:
yes, I wanted to use this and it's a good way to get rid of the config altogether. I didn't do it because of the ip range updates, but I don't see a reason for it anymore. Thanks.